src/charm.py

```diff
@@ -1,104 +1,397 @@
 #!/usr/bin/env python3
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
-#
-# Learn more at: https://juju.is/docs/sdk
-
-"""Charm the service.
-
-Refer to the following post for a quick-start guide that will help you
-develop a new k8s charm using the Operator Framework:
-
-    https://discourse.charmhub.io/t/4208
-"""
 
+import json
 import logging
+import os
+from pathlib import Path
+from typing import Union
 
-from ops.charm import CharmBase
-from ops.framework import StoredState
+import charms.alertmanager_k8s.v0.alertmanager_remote_configurer as remote_config_write
+import yaml
+from charms.observability_libs.v1.kubernetes_service_patch import (
+    KubernetesServicePatch,
+    ServicePort,
+)
+from ops.charm import (
+    CharmBase,
+    ConfigChangedEvent,
+    PebbleReadyEvent,
+    RelationJoinedEvent,
+)
 from ops.main import main
-from ops.model import ActiveStatus
+from ops.model import (
+    ActiveStatus,
+    BlockedStatus,
+    MaintenanceStatus,
+    ModelError,
+    WaitingStatus,
+)
+from ops.pebble import Layer
+
+from config_dir_watcher import (
+    AlertmanagerConfigDirWatcher,
+    AlertmanagerConfigFileChangedCharmEvents,
+    AlertmanagerConfigFileChangedEvent,
+)
 
 logger = logging.getLogger(__name__)
 
 
```

Comment:
What happens when we add units to the application? Does it still work?

Reply:
It does.